remove server selection timeout override for mongodbhandler

bustedware opened the pull request:
## Description

remove server selection timeout override in mongodb handler. 
[default is best/safe at 30000ms](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.serverSelectionTimeoutMS) and can be configured in connection string if needing to override

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: NA / tested locally
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

set serverSelectionTimeoutMS to 0 in connection string and observe failure
set serverSelectionTimeoutMS to 30000 in connection string and observe success
don't set serverSelectionTimeoutMS in connection string and observe success for connections above the previous 5000ms

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

ZoranPandovski commented:
recheckcla

mindsdbadmin commented:
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   **Jacob Bills** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

ZoranPandovski commented:
@ea-rus any comment against this :) ?

bustedware commented:
I have read the CLA Document and I hereby sign the CLA

ea-rus commented:
I think it is fine

ZoranPandovski commented:
@bustedware Please make sure the user you made the commits is added to your GH account https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user.

bustedware commented:
@ZoranPandovski ok im pretty sure i added my email address back now. i had removed it for my commits to golang to work though so not sure what you see exactly. can you check again?

ZoranPandovski commented:
recheckcla
